src/plugin/indexer-kafka/src/java/org/apache/nutch/indexwriter/kafka/KafkaIndexWriter.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//TODO refactor the dependencies out of root ivy file
```

Comment:
@r0ann3l I've spent some time on ivy but I'm no expert. I can't seem to get the project to resolve the dependencies when they are set from the Indexer-Kafka module. I followed the same format that indexer-elastic-rest used and placed the dependencies in the root ivy file so this could all work. 

Can anybody help me resolve this issue?

Reply:
Hi @AyalCiobotaru, I've made some changes to solve your problem. Please, merge them from [here](https://github.com/r0ann3l/nutch/tree/NUTCH-2631). Check this: on `KafkaIndexWriter.open(KafkaIndexWriter.java:75)` I set the context `ClassLoader` for current thread as the `ClassLoader` of the current class. The thing here is the kafka-clients-1.1.0 library tries to load the classes for keySerializer and valueSerializer from context `ClassLoader` of the current thread and this classes are not available from there. I'm not sure if this workaround is correct, but it works for me in local mode and Kafka 1.12. What do you think about it?